common/remote/sockfile.cpp

```diff
@@ -5745,13 +5745,29 @@ class CRemoteFileServer : implements IRemoteFileServer, public CInterface
             }
         }
 
-        if (_useSSL)
+        if (_connectMethod == SSLOnly || _connectMethod == SSLFirst || _connectMethod == UnsecureFirst)
         {
             if (sslep.port == 0)
-                throw createDafsException(DAFSERR_connection_failed, "Secure dafilesrv port not specified");
+                throw createDafsException(DAFSERR_serverinit_failed, "Secure dafilesrv port not specified");
 
-            if ( (_useSSL != UnsecureFirst) && (!securitySettings.certificate || !securitySettings.privateKey) )
-                throw createDafsException(DAFSERR_connection_failed, "SSL Certificate and/or Key file information not found in environment.conf");
+            if (_connectMethod == UnsecureFirst)
```

Comment:
Warning will help diagnose, but I am not that keen on it continuing if it's really a configuration error.
Okay it _could_ continue as it's UnsecureFirst, but I'm not convinced it should.
i.e. shouldn't it be treated as an error, just as if ssl port was missing.